org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -244,18 +325,23 @@ private void handleLinkedFiles(Collection<File> tmpToDelete, final File baseDir)
     }
   }
 
-  public Map<IResource, List<Issue>> run(final Properties props, final SonarRunnerFacade runner, final IProgressMonitor monitor) {
-    SonarLintCorePlugin.getDefault().debug("Start analysis with args:\n" + propsToString(props));
+  public Map<IResource, List<Issue>> run(final AnalysisConfiguration config, final SonarLintClientFacade sonarlint, final IProgressMonitor monitor) {
+    SonarLintCorePlugin.getDefault().debug("Start analysis with configuration:\n" + config.toString());
     Thread.UncaughtExceptionHandler h = exceptionHandler();
     final Map<IResource, List<Issue>> issuesPerResource = new HashMap<>();
     Thread t = new Thread("SonarLint analysis") {
       @Override
       public void run() {
-        runner.startAnalysis(props, new IssueListener() {
+        sonarlint.startAnalysis(config, new IssueListener() {
 
           @Override
           public void handle(Issue issue) {
-            IResource r = ResourceUtils.findResource(request.getProject(), issue.getComponentKey());
+            IResource r;
+            if (issue.getFilePath() == null) {
+              r = request.getProject();
+            } else {
+              r = ResourcesPlugin.getWorkspace().getRoot().getFileForLocation(Path.fromOSString(issue.getFilePath().toString()));
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'issue' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)